youtube_dl/extractor/platzi.py

```diff
@@ -104,50 +104,40 @@ def _real_extract(self, url):
         lecture_id = self._match_id(url)
 
         webpage = self._download_webpage(url, lecture_id)
-
-        data = self._parse_json(
-            self._search_regex(
-                # client_data may contain "};" so that we have to try more
-                # strict regex first
-                (r'client_data\s*=\s*({.+?})\s*;\s*\n',
-                 r'client_data\s*=\s*({.+?})\s*;'),
-                webpage, 'client data'),
+        data_preloaded_state = self._parse_json(
+            self._search_regex((r'window.__PRELOADED_STATE__ = (.*)\<\/script'), webpage, 'client data'),
             lecture_id)
 
-        material = data['initialState']['material']
-        desc = material['description']
-        title = desc['title']
-
+        # desc = data_preloaded_state['videoPlayer']['courseDescription']
```

Comment:
Avoid crashing randomly if JSON items are moved or renamed or otherwise unexpected:
```
        video_player = try_get(data_preloaded_state, lambda x: x['videoPlayer'], dict)
        title = video_player.get('name')
        # if there may be other ways to get the title, try them here, then ...
        if not title:
            raise ExtractorError('No title for page')
        duration = video_player.get('duration')
        formats = []
        for server, server_json in (try_get(video_player, lambda x: x['video']['servers'], dict) or {}).items():
```